runtime/astar/src/lib.rs

```diff
@@ -86,6 +86,50 @@ mod xcm_config;
 
 pub type AstarAssetLocationIdConverter = AssetLocationIdConverter<AssetId, XcAssetConfig>;
 
+pub type AstarPSP34ClassIdConverter = PSP34ClassIdConverter<MultiLocation,ClassId>;
+
+impl<ClassId: Clone, Multilocation:Clone> Convert<ClassId,MultiLocation> for AstarPSP34ClassIdConverter<ClassId,MultiLocation> {
+	
+    fn convert(value: MultiLocation) -> Result<ClassId, MultiLocation> {
+        match (value.parents, value.interior) {
+            (x,Junctions(X1(AccountId32{id: y, ..}))) if x==0 => Ok(y),
+            _ => Err(value)
+        }
+	}
```

Comment:
How would we distinguish between _normal_ accounts and "WASM smart contracts, PSP34 compliant ones, supported by pallet-contracts on Astar"?

To me it seems like the proposed MultiLocation isn't verbose enough or different enough from what's already use.

Reply:
> How would we distinguish between normal accounts and "WASM smart contracts, PSP34 compliant ones, supported by pallet-contracts on Astar"?

I couldn't find any method/work-around in pallet_contract which could verify if a contract is PSP34 or not, so I thought I doesn't make sense to make Multilocation more verbose because it wouldn't make sense if couldn't validate it and it would all be the same.
But you are right, for the sanity of everyone, we should make it more verbose.

Reply:
I don't think there's a way for you to know if contract is PSP34 - best you can do is to _assume_ it is and prepare the appropriate call (input data) for the `pallet-contracts' `bare_call`.


When MultiLocation's are considered, it's important to be very specific.
Imagine a scenario where a new type of NFT standard is published, e.g. PSP12345
Somehow it retains backwards compatibility with PSP34 but has many new extra functionalities.

This would mean we could support BOTH, and based on the MultiLocation referenced, we could _deposit_ NFT to either the PSP34 or PSP12345 contract.